packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -244,7 +246,12 @@ export const Combobox = <T extends string | number>(props: ComboboxProps<T>) =>
     // Instead, stateReducer is called in the same tick as state changes, before that state is committed and rendered.
 
     onSelectedItemChange: ({ selectedItem }) => {
-      onChange(selectedItem);
+      // this is an else if because TS can't do (isClearable || selectedItem !== null)
+      if (isClearable) {
+        onChange(selectedItem);
+      } else if (selectedItem !== null) {
+        onChange(selectedItem);
+      }
```

Comment:
wonder if it makes the logic clearer to be explicit, like:
```tsx
      if (isClearable) {
        onChange(selectedItem);
      } else if {
        if (selectedItem !== null) {
          onChange(selectedItem);
        }
      }
```
wdyt?

Reply:
imo, slightly clearer without the nesting. i did make the comment there a little better though, with the intended condition directly above the if statement and explanation of the type issue.